### ISSUE TITLE
Missing discernible text on `ClayDropDown.Search`

### ISSUE THREAD

**Issue by markocikos** [liferay/clay#5377]
## Is there an example you can provide

Related JIRA ticket: https://issues.liferay.com/browse/LPS-175752

Reproducible in storybook: https://storybook.clayui.com/?path=/story/design-system-components-dropdown--search

![screen](https://user-images.githubusercontent.com/5435511/219046987-fa2f19e2-1717-434a-8c84-98b9537a4bd0.png)

## What are the steps to reproduce?

See steps to reproduce in JIRA ticket. Apply the same use of screen reader to storybook page.

## What is the expected result?

Button has discernible text.

`ClayButton` [on this line](https://github.com/liferay/clay/blob/master/packages/clay-drop-down/src/Search.tsx#L92-L101) should have `aria-label`, or there should be an API to set attributes of this element.
